plugins/hypervisors/hyperv/DotNet/ServerResource/HypervResource/HypervResourceController.cs

```diff
@@ -978,14 +1025,58 @@ public JContainer ModifyStoragePoolCommand([FromBody]dynamic cmd)
                     else if (poolType == StoragePoolType.NetworkFilesystem ||
                         poolType == StoragePoolType.SMB)
                     {
-                        NFSTO share = new NFSTO();
-                        String uriStr = "cifs://" + (string)cmd.pool.host + (string)cmd.pool.path;
-                        share.uri = new Uri(uriStr);
-                        hostPath = Utils.NormalizePath(share.UncPath);
+                        if ((bool)cmd.add)
```

Comment:
this ethod is getting very big. extracting some of the code to a private method would make it more readable (and probably more maintainable.